### PR TITLE
Chore: Cosmetic changes to how we resolve session properties for a model

### PR DESCRIPTION
**Pull request by izeigerman**

### CODE REVIEW

**Review by georgesittas — APPROVED** [review]

File: sqlmesh/core/model/definition.py [hunk 1]
```diff
@@ -1880,21 +1880,20 @@ def _split_sql_model_statements(
...
-    defaults: t.Optional[t.Dict[str, t.Any]],
-    provided: t.Optional[exp.Expression] | t.Optional[t.Dict[str, t.Any]],
+def _resolve_session_properties(
+    default: t.Optional[t.Dict[str, t.Any]],
+    provided: t.Optional[exp.Expression | t.Dict[str, t.Any]],
 ) -> t.Optional[exp.Expression]:
     if isinstance(provided, dict):
         session_properties = {k: exp.Literal.string(k).eq(v) for k, v in provided.items()}
+    elif provided:
+        session_properties = {expr.this.name: expr for expr in provided}
```
Comment (tobymao) [inline sqlmesh/core/model/definition.py:1890]: what is expr? what are you expecting it to be such that expr.this.name works

**Review by tobymao — APPROVED** [review]

File: sqlmesh/core/model/definition.py (hunk 1 above)
Comment (izeigerman) [reply sqlmesh/core/model/definition.py:1890]: It's a tuple of properties.

Comment (izeigerman) [reply sqlmesh/core/model/definition.py:1890]: so we expect this to be `EQ` with the left operand being a string literal or an identifer

Comment (tobymao) [reply sqlmesh/core/model/definition.py:1890]: sg

**Pull request merged**